diff --git a/futurerestore/futurerestore.cpp b/futurerestore/futurerestore.cpp
index 6666dac..191136c 100644
--- a/futurerestore/futurerestore.cpp
+++ b/futurerestore/futurerestore.cpp
@@ -700,7 +700,6 @@ void futurerestore::enterPwnRecovery(plist_t build_identity, std::string bootarg
             //retassure(((recovery_client_new(_client) == IRECV_E_SUCCESS) ||
             //           (mutex_unlock(&_client->device_event_mutex), 0)),
             //          "Failed to connect to device in Recovery Mode!");
-        }
     } else if ((_client->device->chip_id >= 0x8006 && _client->device->chip_id <= 0x8030) ||
                (_client->device->chip_id >= 0x8101 && _client->device->chip_id <= 0x8301)) {
         dfu = true;
