diff --git a/configure.ac b/configure.ac
index 0405e05..d5b08a7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -23,9 +23,9 @@ esac
 
 # Checks for libraries.
 
-PKG_CHECK_MODULES(libplist, libplist >= 1.0)
+PKG_CHECK_MODULES(libplist, libplist-2.0 >= 1.0)
 PKG_CHECK_MODULES(libimobiledevice, libimobiledevice-1.0 >= 1.2.1)
-PKG_CHECK_MODULES(libirecovery, libirecovery >= 0.2.0)
+PKG_CHECK_MODULES(libirecovery, libirecovery-1.0 >= 0.2.0)
 
 
 # Checks for header files.
